test/powershell/Modules/Microsoft.PowerShell.Management/Get-HotFix.Tests.ps1

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "Get-HotFix Tests" -Tag CI {
+    BeforeAll {
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+
+        $skip = $false
+        if (!$IsWindows) {
+            $skip = $true
+        }
+        else {
+            $qfe = Get-CimInstance Win32_QuickFixEngineering
+            if ($qfe.Count -eq 0) {
+                $skip = $true
+            }
+        }
+
+        $PSDefaultParameterValues["it:skip"] = $skip
+    }
+
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    It "Get-HotFix will enumerate all QFEs" {
+        $hotfix = Get-HotFix
+        $hotfix.Count | Should -Be $qfe.Count
+    }
+
+    It "Get-HotFix can filter on -Id" {
+        $testQfe = $qfe[0]
+
+        $hotfix = Get-HotFix -Id $testQfe.HotFixID
+        $hotfix.HotFixID | Should -BeExactly $testQfe.HotFixID
+        $hotfix.Description | Should -BeExactly $testQfe.Description
+    }
+
+    It "Get-HotFix can filter on -Description" {
+        $testQfes = $qfe | Where-Object { $_.Description -eq 'Update' }
+        $hotfixes = Get-HotFix -Description 'Update'
+        $hotfixes.Count | Should -Be $testQfes.Count
+    }
```

Comment:
It seems the test is not reliable - what if a description does not contain "Update"? Also it checks nothing about the cmdlet. I'd remove the test.

Reply:
It would be a big change for updates not to have the work update in them.

Reply:
Filtering is  implemented in Where-Object. Why do we need to test Where-Object? :-)
Also `Description` can be localized.

Reply:
Since the cmdlet exposes a `-Description` parameter, I want to just have some test against it.  I can skip if $testQfes is empty

Reply:
Actually in this case if there isn't any hotfixes with description 'update', then that would be sufficient to pass the test.  On my win7 box where I tested this, most of the hotfixes had a description 'update'

Reply:
> Since the cmdlet exposes a -Description parameter, I want to just have some test against it. 

It is better to have _explicit_ test for the case. Implicit checks can be removed

Reply:
I think this is a valid test, but I can add an `else` condition if no qfes are found with description `Update`